### DISCUSSION TITLE
Field为disabled状态时，调用form.setPattern('readPretty'), 这个Field的显示仍为disable 样式，这是正常的吗？期望为阅读模式

### DISCUSSION THREAD

**Discussion by specialCoder** [alibaba/formily#1740] [Q&A 2.x] [answered]
<Field
        name="costPoint"
        title="补充"
        required
        disabled
        decorator={[FormItem]}
        component={[NumberPicker,{
          precision:0,
          placeholder:'请输入'
        }]}
      />

**Comment by janryWang** [comment]
能不能提问带上复现demo，看到你提了好多问题，一个都没带，这种问题是最低效的

**Reply by specialCoder** [reply]
https://codesandbox.io/s/formilydisabled-bwyq9?file=/App.tsx

**Reply by specialCoder** [reply]
操作步骤： 1. 在文本框里输入任意字符，点击预览，发现disabled的Field不会变成readPretty【不符合预期，期望整体都变成阅读模式】； 2.点击编辑，返回编辑状态，此时输入框是disabled 【符合预期】

**Comment by janryWang** [comment, marked as answer]
字段级别设置的pattern优先级要高于form级别设置的，你要强制设置的话，可以用query查询所有字段，然后批量设置pattern
